### PR TITLE
expression/aggregation: migrate test-infra to testify for aggregation_test.go

### LINKED ISSUE #27906: migrate test-infra to testify for `expression/aggregation/aggregation_test.go`
This is a subtask of #27904

### PR DESCRIPTION
**Pull request by penthaapatel**
Signed-off-by: Penthaa Patel <penthaapatel@gmail.com>

### What problem does this PR solve?

Issue Number: close #27906

### What is changed and how it works?

What's Changed:

- migrate test-infra to testify for `expression/aggregation/aggregation_test.go`

### Release note 

```release-note
None
```

### CODE REVIEW

File: expression/aggregation/aggregation_test.go [hunk 1]
```diff
@@ -46,57 +45,62 @@ func generateRowData() []chunk.Row {
...
 	needed = types.NewDecFromStringForTest("50.500000000000000000000000000000")
-	c.Assert(result.GetMysqlDecimal().Compare(needed) == 0, IsTrue)
+	require.True(t, result.GetMysqlDecimal().Compare(needed) == 0)
 	partialResult := distinctAvgFunc.GetPartialResult(evalCtx)
-	c.Assert(partialResult[0].GetInt64(), Equals, int64(100))
+	require.Equal(t, int64(100), partialResult[0].GetInt64())
 	needed = types.NewDecFromStringForTest("5050")
-	c.Assert(partialResult[1].GetMysqlDecimal().Compare(needed) == 0, IsTrue, Commentf("%v, %v ", result.GetMysqlDecimal(), needed))
+	require.True(t, partialResult[1].GetMysqlDecimal().Compare(needed) == 0)
+	//c.Assert(partialResult[1].GetMysqlDecimal().Compare(needed) == 0, IsTrue, Commentf("%v, %v ", result.GetMysqlDecimal(), needed))
```
Comment (penthaapatel) [inline expression/aggregation/aggregation_test.go:99]: @tisonkun Where to log the following comment? 
`Commentf("%v, %v ", result.GetMysqlDecimal(), needed)`

Comment (tisonkun) [reply expression/aggregation/aggregation_test.go:99]: `c.Assert(partialResult[1].GetMysqlDecimal().Compare(needed) == 0, IsTrue, Commentf("%v, %v ", result.GetMysqlDecimal(), needed))`

can be almost

`require.Equalf(t, 0, partialResult[1].GetMysqlDecimal().Compare(needed), "%v, %v ", result.GetMysqlDecimal(), needed)`

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
@penthaapatel thanks for your contribution, it is clean work!

**Review by xhebox — APPROVED** [review]

**Pull request merged**